slack.go

```diff
@@ -69,6 +74,10 @@ func (c *Client) SendMessage(msg *Message) error {
 	return nil
 }
 
+func (c *Client) SetUrl(url string) {
+	c.Url = url
+}
+
 func (m *Message) NewAttachment() *Attachment {
```

Comment:
and then we can use `NewClient` and return the interface, while passing the url:

```
func NewClient(url string) Client {
    return &client{url}
}
```